### PR TITLE
[JENKINS-51533] Bugfix not showing all artifacts

### PR DESCRIPTION
**Pull request by samrocketman**
Description
-----------

Files which are the same _name_ are not shown as artifacts in the Blue Ocean UI.

It is possible for multiple files of the same name to be collected but under different artifact paths.  This change fixes a conflict of the unique ID field where artifacts with the same name but different paths should have a unique ID.  By having a unique ID, all artifacts will be shown in the Blue Ocean UI.

Compare the following discrepencies of artifacts being listed in these two URLs.

- https://build.gimp.org/job/gimp/job/master/117/ (complete artifact list)
- https://build.gimp.org/blue/organizations/jenkins/gimp/detail/master/117/artifacts (incomplete artifact list)

> **Note:** `test-suite.log` is a file with the same name collected under different paths.

See also:

- [JENKINS-51533][JENKINS-51533] Blue Ocean Artifact tab not showing all artifacts.

Manual Test
-----------

Requirements:

* Docker
* Bash

Bootstrap Jenkins.

```bash
git clone --recursive https://gitlab.gnome.org/Infrastructure/gimp-ci-jenkins.git
cd gimp-ci-jenkins/
./jenkins_bootstrap.sh
```

Open `http://localhost:8080/` and after GIMP master is finished being built, start another build of GIMP master.  The second build of GIMP master is required so that GIMP generates unit tests.  By default pipelines do not run unit tests for the first Jenkins build in order to safely create all dependency artifacts.

Build the Blue Ocean plugin at this pull request commit and upload the following artifacts to Jenkins:

- `blueocean-commons/target/blueocean-commons.hpi` (required by `blueocean-rest.hpi`)
- `blueocean-rest/target/blueocean-rest.hpi` (contains the fix)

Screenshots
-----------

Before

screenshot-before

After

screenshot-after

Submitter checklist
-------------------

- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

[JENKINS-51533]: https://issues.jenkins-ci.org/browse/JENKINS-51533

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by samrocketman** [comment]
@vivek please review.  I need to modify a test you wrote in a class because I'm changing how ArtifactImpl.java IDs are calculated.  You made the change in #1079.

File: blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/rest/ArtifactImplTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@RunWith(PowerMockRunner.class)
+public class ArtifactImplTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
```
[the remark this answers is not in the document]
Comment (samrocketman) [reply blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/rest/ArtifactImplTest.java]: Yeah, I realized I didn't.  I'm not really familiar with mocking so I was trying to copy pasta by referencing other classes with mocks.

**Comment by sgleske-ias** [comment]
@olamy Please add the `needs-review` label to this issue.  It is no longer a WIP and is ready for review.

**Comment by samrocketman** [comment]
Thanks @vivek

**Pull request merged**